src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -457,14 +456,42 @@ final class RiskProvider: RiskProviding {
 	
 	private func checkIfRiskLevelHasChangedForNotifications(_ risk: Risk) {
 		/// Triggers a notification for every risk level change.
-		if risk.riskLevelHasChanged {
-			Log.info("Trigger notification about changed risk level", log: .riskDetection)
-			UNUserNotificationCenter.current().presentNotification(
-				title: AppStrings.LocalNotifications.detectExposureTitle,
-				body: AppStrings.LocalNotifications.detectExposureBody,
-				identifier: ActionableNotificationIdentifier.riskDetection.identifier
-			)
+		switch risk.riskLevelChange {
+		case .decreased:
+			Log.info("decrease risk change state won't trigger a high risk notification")
+		case .increased:
+			triggerHighRiskNotification()
+		case let .unchanged(riskLevel):
+			guard riskLevel == .high,
+				  let mostRecentDateWithRiskLevel = risk.details.mostRecentDateWithRiskLevel,
+				  let previousMostRecentDateWithRiskLevel = store.mostRecentDateWithRiskLevel,
```

Comment:
I think we could get rid of the new stored `mostRecentDateWithRiskLevel` if we just passed in the previous risk here. This func is called right after we initialized the new risk and we would still have the previous calculation result available there to also get the previous risk.